src/mewbot/api/registry.py

```diff
@@ -121,14 +165,48 @@ def do_register(api: Type[Component]) -> Type[Component]:
 
     @classmethod
     def _detect_api_versions(mcs, impl: Type[Any]) -> Iterable[Tuple[ComponentKind, str]]:
+        """Finds all API versions that match a given implementation."""
+
         for kind, apis in mcs._api_versions.items():
             for version, api in apis.items():
                 if issubclass(impl, api):
                     yield kind, version
 
     @classmethod
-    def api_version(mcs, component: Component) -> Tuple[ComponentKind, str]:
-        for val in mcs._detect_api_versions(type(component)):
+    def api_version(mcs, component: Component | Type[Component]) -> Tuple[ComponentKind, str]:
+        """
+        Gets the Component Kind (e.g. 'Behaviour') and API version (e.g. 'v1') from a component.
+
+        This function will only return the first matched type, as per the Registry's
+        'one base type' constraint.
+        """
+
+        if not isinstance(component, type):
+            component = type(component)
+
+        for val in mcs._detect_api_versions(component):
             return val
 
         raise ValueError(f"No API version for {component}")
+
+    @staticmethod
+    def load_and_register_modules(name: Optional[str] = None) -> Iterable[Any]:
+        """
+        Load modules from setuptools that declare a `mewbot-v1` entrypoint.
```

Comment:
Does this comment want to say "mewbot-v1" or "setuptools that declare an entrypoint matching the current api"
```suggestion
        Load modules from setuptools that declare a entrypoint with a valid API.
```

Reply:
Hmm, interesting. If we're going for that, I'd probably say `supported`?

Reply:
Sounds reasonable to me
```suggestion
        Load modules from setuptools that declare an entrypoint with a supported API.
```